src/adm/settings.rs

```diff
@@ -59,12 +63,44 @@ pub struct AdvertiserUrlFilter {
     pub(crate) paths: Option<Vec<PathFilter>>,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum PathMatching {
+    Prefix,
+    Exact,
+}
+
+impl TryFrom<&str> for PathMatching {
+    type Error = ConfigError;
+
+    fn try_from(string: &str) -> Result<Self, Self::Error> {
+        match string.to_lowercase().as_str() {
+            "prefix" => Ok(Self::Prefix),
+            "exact" => Ok(Self::Exact),
+            _ => Err(ConfigError::Message(format!(
+                "Invalid Path Filter Type {}",
+                string
+            ))),
+        }
+    }
+}
+
+impl From<PathMatching> for String {
```

Comment:
total nit because I don't even think this is used but this impl could be `for &'static str` to avoid the allocation.

Reply:
Yeah, I do these out of habit because I inevitably need them later.
Thanks for the tip, though. I'll do that.